app/livechat/server/api/v1/contact.js

```diff
@@ -40,3 +38,31 @@ API.v1.addRoute('omnichannel/contact', { authRequired: true }, {
 		return API.v1.success({ contact });
 	},
 });
+
+
+API.v1.addRoute('omnichannel/contact.search', { authRequired: true }, {
+	get() {
+		try {
+			check(this.queryParams, {
+				email: Match.Maybe(String),
+				phone: Match.Maybe(String),
+			});
+
+			const { email, phone } = this.queryParams;
```

Comment:
```suggestion
			const { email, phone } = this.queryParams;
			if (!email && !phone) {
				throw new Meteor.Error('error-invalid-params');
			}

			const query = Object.assign({},
				...email && { visitorEmails: { address: email } },
				...phone && { phone: { phoneNumber: phone } },
			);
```

Reply:
That solution isn't working:

`message:'undefined is not a function'`

Reply:
The new properties assigned were not inside an object.
It was very simple to fix.